src/components/study-container.js

```diff
@@ -577,49 +577,23 @@ export function StudyContainer({ view, onChangeTab }) {
 
     const checkStudyIndexation = useCallback(() => {
         setIsStudyIndexationPending(true);
-        fetchStudyIndexationStatus(studyUuid)
-            .then((status) => {
-                switch (status) {
-                    case STUDY_INDEXATION_STATUS.INDEXED: {
-                        dispatch(setStudyIndexationStatus(status));
-                        setIsStudyIndexationPending(false);
-                        break;
-                    }
-                    case STUDY_INDEXATION_STATUS.INDEXING_ONGOING: {
-                        dispatch(setStudyIndexationStatus(status));
-                        break;
-                    }
-                    case STUDY_INDEXATION_STATUS.NOT_INDEXED: {
-                        dispatch(setStudyIndexationStatus(status));
-                        reindexAllStudy(studyUuid)
-                            .catch((error) => {
-                                // unknown error when trying to reindex study
-                                snackError({
-                                    headerId: 'studyIndexationError',
-                                    messageTxt: error,
-                                });
-                            })
-                            .finally(() => {
-                                setIsStudyIndexationPending(false);
-                            });
-                        break;
-                    }
-                    default: {
-                        setIsStudyIndexationPending(false);
-                        snackError({
-                            headerId: 'studyIndexationStatusUnknown',
-                            headerValues: { status: status },
-                        });
-                        break;
-                    }
-                }
+        reindexStudyIfNeeded(studyUuid)
+            .then(() => {
+                dispatch(
```

Comment:
Not really INDEXING_ONGOING, maybe we should dispatch the status only on notification consumption.  Then remove all in this then...